src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -48,125 +39,163 @@
+  private static final String
+      CLOSE_BUTTON_SELECTOR_FORMAT = SLOT_SELECTOR_PREFIX + CLOSE_BUTTON_CLASS_NAME;
 
   private static final int EXPECTED_PERCENTAGE_DIFFERENCE_IN_VIDEO_AD_HEIGHT = 40;
 
+  private static final String URL_FIRSTQUARTILE = "ad_vast_point=firstquartile";
+  private static final String URL_MIDPOINT = "ad_vast_point=midpoint";
+
   private final WikiaWebDriver driver;
 
   private final Wait wait;
 
   private final String slot;
 
-  private final String videoIframeSelector;
+  private final By imaBridgeSelector;
+  private final By adIframeSelector;
+  private final By speakerSelector;
+  private final By pauseOverlaySelector;
+  private final By closeButtonSelector;
+  private final By progressBarSelector;
+  private final By replayOverlaySelector;
 
-  private boolean playing;
+  private boolean mobile;
 
-  private boolean muted;
+  public AutoplayVuap(WikiaWebDriver driver, String slot, String adIframeId) {
+    this(driver, slot, By.id(adIframeId), false);
+  }
 
-  public AutoplayVuap(WikiaWebDriver driver, String slot, String videoIframeSelector) {
+  public AutoplayVuap(WikiaWebDriver driver, String slot, By adIframeSelector, Boolean mobile) {
     this.driver = driver;
     this.wait = new Wait(driver);
 
     this.slot = slot;
-    this.videoIframeSelector = videoIframeSelector;
-    this.playing = true;
-    this.muted = true;
+    this.imaBridgeSelector = By.cssSelector("#" + slot + " .video-player iframe[src*='imasdk']");
+    this.adIframeSelector = adIframeSelector;
+    this.mobile = mobile;
+
+    speakerSelector = By.cssSelector(String.format(SPEAKER_SELECTOR_FORMAT, slot));
+    pauseOverlaySelector = By.cssSelector(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
+    closeButtonSelector = By.cssSelector(String.format(CLOSE_BUTTON_SELECTOR_FORMAT, slot));
+    progressBarSelector = By.cssSelector(String.format(CURRENT_TIME_SELECTOR_FORMAT, slot));
+    replayOverlaySelector = By.cssSelector(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
   }
 
   public void mute() {
-    if (!muted) {
-      clickElement(String.format(SPEAKER_SELECTOR_FORMAT, slot));
-      muted = true;
+    if (!isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
     }
   }
 
   public void unmute() {
-    if (muted) {
-      clickElement(String.format(SPEAKER_SELECTOR_FORMAT, slot));
-      muted = false;
+    if (isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
     }
   }
 
-  public void pause() {
-    if (playing) {
-      clickElement(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
-      playing = false;
-    }
+  public void togglePause() {
+    wait.forElementClickable(pauseOverlaySelector).click();
   }
 
   public void play() {
-    if (!playing) {
-      clickElement(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
-      playing = true;
+    if (isPaused()) {
+      togglePause();
+    } else if( !this.mobile ) {
+      clickOnArea(2);
```

Comment:
I'm not sure about consts like SECOND_AREA, it will be "magic" anyway. It should be called something like AREA_TO_TRIGGER_PLAY, but it depends on ad, so maybe it should be in provider